### PR TITLE
Add namespace lifecycle to networking.go

### PR DESCRIPTION
**Pull request by jayunit100**
This will be generalized in #6553 , for now this makes it so networking.go does the _right thing_, i.e. creates a namespace specific to the test, inside the test API itself, so that it doesnt rely on the  `admission_control=NamespaceLifecycle,NamespaceAutoProvision,AlwaysAdmit"` parameters from #6539 issue, which otherwise makes this test kind of dangerous to run

### CODE REVIEW

File: test/e2e/networking.go [hunk 1]
```diff
@@ -25,10 +25,18 @@ import (
...
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
+//Namespace constants, implemented
+var svcname = "nettest"
+var ns = svcname + "-" + randomSuffix()
+var c *client.Client = nil
+var nso *api.Namespace
```
Comment (quinton-hoole) [inline test/e2e/networking.go]: nit: Prefer a more descriptive variable name here.

**Comment by quinton-hoole** [comment]
Other than the minor nit, LGTM.  Please post the results of a few successful runs of this test here before merging.

**Comment by jayunit100** [comment]
okay ! testing now.

**Comment by jayunit100** [comment]
```
Running Suite: Kubernetes e2e Suite run 1 of 1
==============================================
Random Seed: 1428533964 - Will randomize all specs
Will run 2 of 37 specs

STEP: Executing a successfull http request from the external internet
STEP: Building a namespace api object
STEP: Creating a kubernetes client
>>> testContext.AuthConfig: /opt/kube-creds
STEP: Creating a namespace for this test suite
&{{ } {nettest-9763      0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} {[]} {}} 

  &{{ } {      0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} {[]} {}} 

************** 
&{%!s(*client.RESTClient=&{0xc2081662a0 v1beta3 false 0xc20803a0a0 0xc20815a2d0 0 0xc20816c160})}
**************SSSSSSSSSSSSSSS
------------------------------
Networking 
  should provide unchanging, static URL paths for kubernetes api services.
  /home/jay/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:145
[It] should provide unchanging, static URL paths for kubernetes api services.
  /home/jay/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:145
STEP: testing: /validate
STEP: testing: /healthz
•SSSSSSSSSSSSSSS
------------------------------
Networking 
  should function for intra-pod communication
  /home/jay/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:276
[It] should function for intra-pod communication
  /home/jay/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:276
STEP: Creating a service named [nettest] in namespace nettest-2414
STEP: Creating a webserver (pending) pod on each node
INFO: Creating pod nettest-0 on node 127.0.0.1
STEP: Waiting for the webserver pods to transition to Running state
STEP: waiting up to 5m0s for pod nettest-0 status to be running
INFO: Waiting for pod nettest-0 in namespace nettest-2414 status to be "running" (found "Pending") (5.565546ms)
STEP: Waiting for connectivity to be verified
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.548647ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.822754ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.509225ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 2.686556ms
INFO: Attempt 3/60: test still running
INFO: About to make a proxy status call
INFO: Proxy status call returned in 2.516342ms
INFO: Passed on attempt 4. Cleaning up.
STEP: Cleaning up the webserver pods
STEP: Cleaning up the service

• [SLOW TEST:15.114 seconds]
Networking
/home/jay/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:278
  should function for intra-pod communication
  /home/jay/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:276
------------------------------
SSSSSSTEP: Destroying namespace for this suite

Ran 2 of 37 Specs in 15.394 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 35 Skipped I0408 18:59:40.345920   27049 driver.go:114] All tests pass
[jay@rhbd kubernetes]$ 

```

**Comment by quinton-hoole** [comment]
Can be merged as soon as Travis or Shippable are happy.

**Comment by jayunit100** [comment]
moving on to #6553 , assuming this is ready to merge now !  thanks for the  real-time review !

**Pull request merged**